hazelcast/src/main/java/com/hazelcast/sql/impl/client/SqlClientService.java

```diff
@@ -104,47 +105,56 @@ public SqlResult execute(@Nonnull SqlStatement statement) {
 
             ClientInvocationFuture future = invokeAsync(requestMessage, connection);
 
-            future.whenComplete(withTryCatch(logger,
-                    (message, error) -> handleExecuteResponse(connection, res, message, error))).get();
+            try {
+                future.whenComplete((message, error) ->  {
+                    if (error != null) {
+                        RuntimeException err = rethrow(error, connection);
```

Comment:
Do you think it would be worth moving this try-catch logic `handleExecuteResponse` as it was before? Feels like this logic is relevant to response only. 
Also not sure, but would it make sense to try to reduce number of try-catch layers here? They're all operating on `Exception` anyway, so its not like something can actually leave the outer try-catch.

Reply:
Yeah definitely I am not very familiar with exception handling in Java client (nodejs client dev here) and that part can be improved. Let me see...

Reply:
Ok changed it a little bit in the way you said.